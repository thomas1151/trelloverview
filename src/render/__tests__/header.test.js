import React from "react";
import ReactDOM from "react-dom";


import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderHeaderFn from "../header";
import { shallowToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('Header Rendering function', () => {
    it("renders with no valid stage", () => {
        let card = { labels: [], color: '000000' }
        let stages = [];
        let stage = '';
        let stageAsTitle = '';
        let setStage = '';
        let logOut = '';
        let restart = '';
        expect(shallowToJson(shallow(renderHeaderFn(stages, stage, setStage, stageAsTitle, logOut, restart)))).toMatchSnapshot();
    });
    it("renders with a valid stage which is initial stage", () => {
        let card = { labels: [], color: '000000' }
        let stages = ['testInitialStage', 'testStage'];
        let stage = 'testInitialStage';
        let stageAsTitle = '';
        let setStage = '';
        let logOut = '';
        let restart = '';
        expect(shallowToJson(shallow(renderHeaderFn(stages, stage, setStage, stageAsTitle, logOut, restart)))).toMatchSnapshot();
    });

    it("renders with a valid stage which is not the initial stage", () => {
        let card = { labels: [], color: '000000' }
        let stages = ['testInitialStage', 'testStage'];
        let stage = 'testStage';
        let stageAsTitle = ''; 
        let setStage = '';
        let logOut = '';
        let restart = '';
        expect(shallowToJson(shallow(renderHeaderFn(stages, stage, setStage, stageAsTitle, logOut, restart)))).toMatchSnapshot();
    });
    
    // it("calls the logout function and returns" () => {
    //     let card = { labels: [], color: '000000' }
    //     let stages = ['testInitialStage', 'testStage'];
    //     let stage = 'testStage';
    //     let stageAsTitle = '';
    //     let setStage = '';
    //     let logOut = '';
    //     let restart = '';
    //     expect(shallowToJson(shallow(renderHeaderFn(stages, stage, setStage, stageAsTitle, logOut, restart)))).toMatchSnapshot();
    // });
})
