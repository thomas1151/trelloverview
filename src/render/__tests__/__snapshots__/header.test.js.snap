// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering function renders with a valid second stage 1`] = `
<header
  className="bg-blue-500 px-4 flex w-full"
>
  <h1
    className="text-lg font-bold text-left text-white flex-1 self-center"
  />
  <div
    className="flex flex-wrap  px-2"
  >
    <div
      className="button-wrap"
    >
      <button
        className="text-white border-white hover:bg-blue-700 font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Logout
      </button>
      <button
        className="text-white border-white hover:bg-blue-700 font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Restart
      </button>
    </div>
  </div>
</header>
`;

exports[`Header Rendering function renders with a valid stage which is initial stage 1`] = `
<header
  className="bg-blue-500 px-4 flex w-full"
>
  <h1
    className="text-lg font-bold text-left text-white flex-1 self-center"
  />
  <div
    className="flex flex-wrap  px-2"
  />
</header>
`;

exports[`Header Rendering function renders with a valid stage which is not the initial or secondary stage 1`] = `
<header
  className="bg-blue-500 px-4 flex w-full"
>
  <h1
    className="text-lg font-bold text-left text-white flex-1 self-center"
  />
  <div
    className="flex flex-wrap  px-2"
  >
    <div
      className="button-wrap"
    >
      <button
        className="text-white border-white hover:bg-blue-700 font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Logout
      </button>
      <button
        className="text-white border-white hover:bg-blue-700 font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Restart
      </button>
      <button
        className="text-white border-white hover:bg-blue-700  font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Back
      </button>
    </div>
  </div>
</header>
`;

exports[`Header Rendering function renders with a valid stage which is not the initial stage 1`] = `
<header
  className="bg-blue-500 px-4 flex w-full"
>
  <h1
    className="text-lg font-bold text-left text-white flex-1 self-center"
  />
  <div
    className="flex flex-wrap  px-2"
  >
    <div
      className="button-wrap"
    >
      <button
        className="text-white border-white hover:bg-blue-700 font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Logout
      </button>
      <button
        className="text-white border-white hover:bg-blue-700 font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Restart
      </button>
      <button
        className="text-white border-white hover:bg-blue-700  font-semibold py-1 px-4 border hover:border-blue-500 hover:border-transparent rounded mx-2 my-1"
        onClick={[Function]}
      >
        Back
      </button>
    </div>
  </div>
</header>
`;

exports[`Header Rendering function renders with no valid stage 1`] = `
<header
  className="bg-blue-500 px-4 flex w-full"
>
  <h1
    className="text-lg font-bold text-left text-white flex-1 self-center"
  />
  <div
    className="flex flex-wrap  px-2"
  />
</header>
`;
